pkg/services/ngalert/schedule/schedule.go

```diff
@@ -487,9 +508,24 @@ func (sch *schedule) ruleRoutine(grafanaCtx context.Context, key ngmodels.AlertR
 				}
 			}()
 		case <-grafanaCtx.Done():
-			// clean up the state only if the reason for stopping the evaluation loop is that the rule was deleted
-			if errors.Is(grafanaCtx.Err(), errRuleDeleted) {
-				clearState()
+			// clean up the state only if the reason for stopping the evaluation loop is that the rule was deleted/paused
+			if errors.Is(grafanaCtx.Err(), errRuleDeleted) || errors.Is(grafanaCtx.Err(), errRulePaused) {
+				reason := ngmodels.StateReasonDeleted
+				if errors.Is(grafanaCtx.Err(), errRulePaused) {
+					reason = ngmodels.StateReasonPaused
+				}
+				// The context dead\ine is 5 seconds which should be enough time for the state to be deleted from the
+				// database and the historian to record the state changes async. Do not cancel the context with defer as
+				// otherwise the context will be canceled before the historian can write any state changes.
+				ctx, cancelFunc := context.WithTimeout(context.Background(), 60*time.Second)
```

Comment:
In the comment the deadline is 5 seconds, not 60 seconds. A minute is quite long don't you think?

Reply:
That is what @yuri-tceretian  suggested. Could you explain?